src/init/features/dataconnect/sdk.ts

```diff
@@ -104,103 +101,90 @@ async function askQuestions(setup: Setup, config: Config): Promise<SDKInfo> {
   if (targetPlatform === Platform.IOS) {
-    const outputDir =
-      newConnectorYaml.generate.swiftSdk?.outputDir ||
-      path.relative(connectorInfo.directory, path.join(appDir, `generated/swift`));
-    const pkg =
-      newConnectorYaml.generate.swiftSdk?.package ??
-      upperFirst(camelCase(newConnectorYaml.connectorId));
-    const swiftSdk = { outputDir, package: pkg };
-    newConnectorYaml.generate.swiftSdk = swiftSdk;
-    displayIOSWarning = true;
+    const swiftSdk = {
+      outputDir: path.relative(connectorDir, path.join(appDir, `dataconnect-generated/swift`)),
+      package: upperFirst(camelCase(connectorYaml.connectorId)),
+    };
+    connectorYaml.generate.swiftSdk = swiftSdk;
   }
 
   if (targetPlatform === Platform.WEB) {
-    const outputDir =
-      newConnectorYaml.generate.javascriptSdk?.outputDir ||
-      path.relative(
-        connectorInfo.directory,
-        path.join(appDir, `generated/javascript/${newConnectorYaml.connectorId}`),
-      );
-    const pkg =
-      newConnectorYaml.generate.javascriptSdk?.package ??
-      `@firebasegen/${connectorInfo.connectorYaml.connectorId}`;
-
     const javascriptSdk: JavascriptSDK = {
-      outputDir,
-      package: pkg,
-      packageJsonDir: newConnectorYaml.generate.javascriptSdk?.packageJsonDir,
+      outputDir: path.relative(
+        connectorDir,
+        path.join(appDir, `dataconnect-generated/js/${connectorYaml.connectorId}`),
+      ),
+      package: `@firebasegen/${connectorYaml.connectorId}`,
+      // If appDir has package.json, Emulator would add Generated JS SDK to `package.json`.
+      // Otherwise, emulator would ignore it. Always add it here in case `package.json` is added later.
+      // TODO: Explore other platforms that can be automatically installed. Dart? Android?
+      packageJsonDir: path.relative(connectorDir, appDir),
     };
-
-    if (
-      (await directoryHasPackageJson(appDir)) &&
-      (await confirm({
-        message: "Would you like to add a dependency on the generated SDK to your package.json?",
-      }))
-    ) {
-      javascriptSdk.packageJsonDir = path.relative(connectorInfo.directory, appDir);
-    }
-    newConnectorYaml.generate.javascriptSdk = javascriptSdk;
+    connectorYaml.generate.javascriptSdk = javascriptSdk;
   }
 
-  if (targetPlatform === Platform.DART) {
-    const outputDir =
-      newConnectorYaml.generate.dartSdk?.outputDir ||
-      path.relative(
-        connectorInfo.directory,
-        path.join(appDir, `generated/dart/${newConnectorYaml.connectorId}`),
-      );
-    const pkg = newConnectorYaml.generate.dartSdk?.package ?? newConnectorYaml.connectorId;
+  if (targetPlatform === Platform.FLUTTER) {
     const dartSdk: DartSDK = {
-      outputDir,
-      package: pkg,
+      outputDir: path.relative(
+        connectorDir,
+        path.join(appDir, `dataconnect-generated/dart/${snakeCase(connectorYaml.connectorId)}`),
+      ),
+      package: snakeCase(connectorYaml.connectorId),
```

Comment:
@maneesht brought this up in chat.

`default` isn't a valid dart package name, but out default connector name is `default`. Need a better pattern here.